M5/code-reviews/widgetsController.js

```diff
@@ -0,0 +1,68 @@
+var express = require("express");
+var widgets = express.Router();
+var widgetsArray = require("../models/widget");
+
+/* code tested fine in Postman, but won't pass test POST, commented out for now
+const validateData = (req, res, next) => {
+    const checkInputDataType = (type, objContainsSameType) => {
+        for (const key in objContainsSameType) {
+            if (typeof objContainsSameType[key] !== type)
+                return { passed: false, invalid: key };
+        }
+        return { passed: true };
+    }
+
+    if (Object.keys(req.body).length !== 5)
+        return res.status(400).send("There are missing or additional pairs(key/value) in your data. Please submit again with correct pairs.");
+
+    const { captainName, title, post, mistakesWereMadeToday, daysSinceLastCrisis } = req.body;
+    let validateResult = null;
+
+    validateResult = checkInputDataType("string", { captainName, title, post });
+    if (!validateResult.passed) 
+        return res.status(400).send(`Opps, either ${validateResult.invalid} is missing or its value is not a string`);
+        
+    validateResult = checkInputDataType("boolean", { mistakesWereMadeToday });
+    if (!validateResult.passed)
+        return res.status(400).send(`Opps, ${validateResult.invalid} is missing or its value is not a boolean`);
+
+    validateResult = checkInputDataType("number", { daysSinceLastCrisis });
+    if (!validateResult.passed)
+        return res.status(400).send(`Opps, ${validateResult.invalid} is missing or its value is not a number`);
+
+    next();
+}
+*/
+
```

Comment:
extract all validation logic into a separate file to hold all reusable "utility" code, e.g. `/Utils/helper.js`